flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/KeyedStream.scala

```diff
@@ -201,15 +201,18 @@ class KeyedStream[T, K](javaStream: KeyedJavaStream[T, K]) extends DataStream[T]
       * @return Returns a DataStream
       */
     @PublicEvolving
-    def process[OUT](processJoinFunction: ProcessJoinFunction[IN1, IN2, OUT]): DataStream[OUT] = {
+    def process[OUT: TypeInformation](
+        processJoinFunction: ProcessJoinFunction[IN1, IN2, OUT])
+      : DataStream[OUT] = {
+
```

Comment:
I would suggest to have a specific var for the type info, sth like:
`val outType : TypeInformation[OUT] = implicitly[TypeInformation[OUT]]`
so that it stands out.